framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1569,7 +1569,7 @@ protected void warnRedundantAnnotations(Tree tree, AnnotatedTypeMirror type) {
         AnnotatedTypeMirror defaultAtms = atypeFactory.getDefaultAnnotations(tree, type);
         for (AnnotationMirror explicitAnno : explicitAnnos) {
             AnnotationMirror defaultAtm = defaultAtms.getAnnotationInHierarchy(explicitAnno);
-            if (AnnotationUtils.areSame(defaultAtm, explicitAnno)) {
+            if (defaultAtm != null && AnnotationUtils.areSame(defaultAtm, explicitAnno)) {
```

Comment:
```suggestion
            // defaultAtm can be null for type variable uses, which don't have a default annotation.
            if (defaultAtm != null && AnnotationUtils.areSame(defaultAtm, explicitAnno)) {
```

Reply:
Do you think this comment helps?
Would it be enough to check once outside of the loop, something like

````
if (defaultAtms.getAnnotations().isEmpty()) {
    // Type variable uses don't have a default annotation. Any explicit annotation is not redundant.
    return;
}
````

Reply:
Yes, I think it's a good improvement.